clients/python/src/model_registry/core.py

```diff
@@ -2,45 +2,109 @@
 
 from __future__ import annotations
 
-from ml_metadata.proto import MetadataStoreClientConfig
+import os
+from dataclasses import dataclass
+from pathlib import Path
+from warnings import warn
+
+import grpc
 
 from .exceptions import StoreException
 from .store import MLMDStore, ProtoType
 from .types import ListOptions, ModelArtifact, ModelVersion, RegisteredModel
 from .types.base import ProtoBase
 from .types.options import MLMDListOptions
+from .utils import header_adder_interceptor
 
 
+@dataclass
 class ModelRegistryAPIClient:
     """Model registry API."""
 
-    def __init__(
-        self,
+    store: MLMDStore
+
+    @classmethod
+    def secure_connection(
+        cls,
+        server_address: str,
+        port: int = 443,
+        user_token: bytes | None = None,
+        custom_ca: bytes | None = None,
+    ) -> ModelRegistryAPIClient:
+        """Constructor.
+
+        Args:
+            server_address: Server address.
+            port: Server port. Defaults to 443.
+            user_token: The PEM-encoded user token as a byte string. Defaults to envvar KF_PIPELINES_SA_TOKEN_PATH.
+            custom_ca: The PEM-encoded root certificates as a byte string. Defaults to envvar CERT.
+        """
+        if not custom_ca:
+            ca_cert = os.environ.get("CERT")
+            if not ca_cert:
+                msg = "CA certificate must be provided"
+                raise StoreException(msg)
+            root_certs = Path(ca_cert).read_bytes()
+        else:
+            root_certs = custom_ca
+        chan_creds = grpc.ssl_channel_credentials(root_certs)
+
+        if not user_token:
+            # /var/run/secrets/kubernetes.io/serviceaccount/token
+            sa_token = os.environ.get("KF_PIPELINES_SA_TOKEN_PATH")
+            if sa_token:
+                user_token = Path(sa_token).read_bytes()
+            else:
+                warn("User access token is missing", stacklevel=2)
+
+        if user_token:
+            chan_creds = grpc.composite_channel_credentials(
+                chan_creds,
+                grpc.access_token_call_credentials(user_token),
+            )
+
+        if port != 443:
```

Comment:
Based on the logic the code will never reach here!

Reply:
the user can build an API service without using the high-level client. that's what I'm also doing on the Robot tests setup. Equally could have built a secure connection on another port.